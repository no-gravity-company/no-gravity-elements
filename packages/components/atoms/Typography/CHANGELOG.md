# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.1.0](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/typography@1.0.22...@no-gravity-elements/typography@1.1.0) (2024-11-12)

### Features

- new button styles ([c850785](https://github.com/no-gravity-company/no-gravity-elements/commit/c850785418de3a4f9ca393c75260ed407061314a))

## [1.0.22](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/typography@1.0.21...@no-gravity-elements/typography@1.0.22) (2024-03-30)

**Note:** Version bump only for package @no-gravity-elements/typography

## [1.0.21](https://github.com/no-gravity-company/no-gravity-elements/compare/@no-gravity-elements/typography@1.0.16...@no-gravity-elements/typography@1.0.21) (2024-03-29)

### Bug Fixes

- Ignore changelogs in lerna config ([9f9c7ee](https://github.com/no-gravity-company/no-gravity-elements/commit/9f9c7ee07e4e05f8dfe9c934bf884515ee8d0732))
