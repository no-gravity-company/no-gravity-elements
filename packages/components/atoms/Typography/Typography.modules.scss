@import '../../../stylesheets/base/typography';

:host {
  color: var(--color, black);
  font-family: 'Noto Sans', sans-serif;
  font-optical-sizing: auto;
  font-style: normal;
  font-variation-settings: 'wdth' 100;
  & > * {
    margin: 0;
  }
}
