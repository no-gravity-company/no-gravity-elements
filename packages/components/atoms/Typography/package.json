{
  "name": "@no-gravity-elements/typography",
  "version": "1.0.21",
  "description": "No Gravity Element",
  "author": "No Gravity Company",
  "private": false,
  "homepage": "https://github.com/no-gravity-company/no-gravity-elements#readme",
  "license": "ISC",
  "main": "lib/index.js",
  "typings": "types/index.ts",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib/index.js",
    "lib/index.js.map",
    "types/index.ts"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/no-gravity-company/no-gravity-elements.git"
  },
  "scripts": {
    "publish-component": "yarn publish --access=public"
  },
  "bugs": {
    "url": "https://github.com/no-gravity-company/no-gravity-elements/issues"
  },
  "gitHead": "2bca302d24e66729aa0b3f42b064554df1a3122f"
}
