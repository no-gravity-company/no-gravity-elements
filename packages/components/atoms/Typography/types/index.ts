import { TypographySizes } from 'packages/types';

export interface TypographyProps {
  tag: string;
  size: TypographySizes;
}
