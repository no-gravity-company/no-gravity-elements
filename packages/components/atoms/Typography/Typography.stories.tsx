import { Meta, StoryObj } from '@storybook/web-components';
import { html } from 'lit-html';

import '@no-gravity-elements/typography';
import { TypographySizes } from '@no-gravity-elements/types';

import { TypographyProps } from './types';

type CustomArgs = TypographyProps & { text: string };

const meta: Meta<CustomArgs> = {
  title: 'atoms/Typography',
  component: 'nge-typography',
  parameters: {
    webComponents: {
      tagName: 'nge-typography',
    },
  },
  argTypes: {
    text: {
      description: 'Custom text',
      control: { type: 'text' },
    },
    tag: {
      description: 'HTML tag used for the typography',
      control: { type: 'text' },
      table: {
        defaultValue: { summary: 'span' },
      },
    },
    size: {
      description: 'Typography size',
      options: Object.values(TypographySizes),
      table: {
        defaultValue: { summary: TypographySizes.p },
      },
    },
  },
  render: ({ tag, text, size }) =>
    html` <nge-typography tag="${tag}" size="${size}">${text}</nge-typography> `,
};

export default meta;
type Story = StoryObj<CustomArgs>;

export const Default: Story = {
  args: {
    tag: 'h4',
    text: 'Example typography',
    size: TypographySizes.h1,
  },
};
