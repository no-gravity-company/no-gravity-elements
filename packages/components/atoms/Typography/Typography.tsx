import { Fragment, FunctionComponent, h, VNode } from 'preact';

import { TypographyProps } from '@atoms/Typography/types';

import { TypographySizes } from '@types';

/**
 * <nge-typography>
 *
 *  Typography component
 *
 * @element nge-typography
 *
 * @prop {h1|h2|h3|h4|h5|h6|h7|sp|p|lp|legal|button} [size] - Size of the text
 * @prop {string} [tag] - Tag that will be rendered inside the Shadow DOM
 *
 * @example
 * <nge-typography tag="h4">Typography</nge-typography>
 */

const Typography: FunctionComponent<TypographyProps> = ({ size, tag }: TypographyProps) => {
  const typographyTag = tag || 'span';

  const getTypographyNode = (): VNode => {
    return h(
      typographyTag,
      { class: (size && TypographySizes[size]) || TypographySizes.p },
      <slot />,
    );
  };

  return <Fragment>{getTypographyNode()}</Fragment>;
};

Typography.observedAttributes = ['tag', 'size'];

export default Typography;
