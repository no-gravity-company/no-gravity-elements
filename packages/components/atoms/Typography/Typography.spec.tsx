import { h } from 'preact';
import { shallow } from 'enzyme';

import Typography from '@atoms/Typography/Typography';

import { TypographySizes } from '@types';

describe('Typography', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Typography size={TypographySizes.h1} tag={'h1'}>
        Test
      </Typography>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it.each`
    tag          | size
    ${'h1'}      | ${TypographySizes.h1}
    ${'h2'}      | ${TypographySizes.h4}
    ${undefined} | ${undefined}
  `('should render correctly with the provided size and tag', ({ tag, size }) => {
    const wrapper = shallow(
      <Typography size={size} tag={tag}>
        Test
      </Typography>,
    );
    const element = wrapper.find(tag || 'span');
    expect(element).not.toBe(null);
    expect(element.hasClass(size || 'p')).toBeTruthy();
  });
});
