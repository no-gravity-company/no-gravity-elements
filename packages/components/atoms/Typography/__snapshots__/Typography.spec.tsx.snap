// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should match the snapshot 1`] = `
<h1
  className="h1"
>
  <slot />
</h1>
`;
